driver/src/main/java/com/code_intelligence/jazzer/Jazzer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer;
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+import com.code_intelligence.jazzer.driver.Driver;
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * Entrypoint for Jazzer to run in a user-controlled JVM rather than the JVM started by the native
+ * Jazzer launcher.
+ *
+ * <p>Arguments to Jazzer are passed as {@code jazzer.*} system properties. For example, setting
+ * the property {@code jazzer.target_class} to {@code com.example.FuzzTest} is equivalent to passing
+ * the argument {@code --target_class=com.example.FuzzTest} to the native launcher.
+ *
+ * <p>Arguments to libFuzzer are passed as command-line arguments.
+ */
+public class Jazzer {
+  public static void main(String[] args) throws IOException {
+    List<String> libFuzzerArgs =
+        Stream.concat(Stream.of(prepareArgv0()), Arrays.stream(args)).collect(toList());
+    System.exit(Driver.start(libFuzzerArgs));
+  }
+
+  private static final FileAttribute<?>[] LAUNCHER_SCRIPT_ATTRIBUTES = isPosix()
```

Comment:
Any particular reason for the fields to be `static`? I would tend to move them into `prepareArgv0`.

Reply:
They are true `static final` constants (since `isPosix()` is as well). I could move them into `prepareArgv0`, but wouldn't that somewhat obscure this fact? I will definitely move them to the end of the class.

Reply:
True, but they are only needed once inside `prepareArgv0` and there's no need to keep them around. But, as mentioned, only a minor difference in taste.

Reply:
I made them local.